### PR TITLE
feat: setup authentication

### PR DESCRIPTION
**Pull request by Kojo-Brown**

### CODE REVIEW

File: src/app/auth/error.interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { HttpErrorResponse, HttpInterceptorFn } from '@angular/common/http';
+import { catchError, throwError } from 'rxjs';
+
+export const errorInterceptor: HttpInterceptorFn = (req, next) => {
+  return next(req).pipe(
+    catchError((error: unknown) => {
+      if (error instanceof HttpErrorResponse) {
+        if (error.status === 401) {
+          console.error('Unauthorized request:', error);
```
Comment (ghost) [inline src/app/auth/error.interceptor.ts:9]: You will probably have to implement refreshing of tokens here. You can do that later

File: src/app/auth/error.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        } else {
+          console.error('HTTP error:', error);
+        }
+      } else {
+        console.log('An error occured', error);
```
Comment (ghost) [inline src/app/auth/error.interceptor.ts:14]: I'm not sure if we need this part

File: src/app/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor() {}
+
+  public isAuthenticated(): boolean {
+    return !!localStorage.getItem('token');
```
Comment (ghost) [inline src/app/services/auth.service.ts]: IMHO `localStorage` should only be used for persistence of state not called directly 
image
We can do something like so ensure it is also reactive

File: src/app/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  }
+
+  public getToken(): string | null {
+    return localStorage.getItem('token');
```
Comment (ghost) [inline src/app/services/auth.service.ts]: Same here

File: src/app/auth/error.interceptor.ts (hunk 2 above)
Comment (Kojo-Brown) [reply src/app/auth/error.interceptor.ts:14]: Probably catching any other exceptional errors?

File: src/app/services/auth.service.ts [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
   public isAuthenticated(): boolean {
-    return !!localStorage.getItem('token');
-  }
+    if (!this.loggedInUser()) {
+      return false;
+    }
 
-  public getToken(): string | null {
-    return localStorage.getItem('token');
+    return true;
```
Comment (Kojo-Brown) [inline src/app/services/auth.service.ts]: This is half baked cos at this point user is not fully authenticated, it is when they are verified before it's confirmed so I will add that layer to the authentication and check for both instances when I get there

**Review by ghost — APPROVED** [review]

**Pull request merged**